server/src/main/java/io/druid/server/http/LookupCoordinatorResource.java

```diff
@@ -284,4 +295,346 @@ public Response getSpecificTier(
       return Response.serverError().entity(ServletResourceUtils.sanitizeException(e)).build();
     }
   }
+
+  @GET
+  @Produces({MediaType.APPLICATION_JSON})
+  @Path("/status")
+  public Response getAllLookupsStatus(
+      @QueryParam("detailed") boolean detailed
+  )
+  {
+    try {
+      Map<String, Map<String, LookupExtractorFactoryMapContainer>> configuredLookups = lookupCoordinatorManager.getKnownLookups();
+      if (configuredLookups == null) {
+        return Response.status(Response.Status.NOT_FOUND)
+                       .entity("No lookups found")
+                       .build();
+      }
+
+      Map<HostAndPort, LookupsState<LookupExtractorFactoryMapContainer>> lookupsStateOnNodes = lookupCoordinatorManager.getLastKnownLookupsStateOnNodes();
+
+      Map<String, Map<String, LookupStatus>> result = new HashMap<>();
+
+      for (Map.Entry<String, Map<String, LookupExtractorFactoryMapContainer>> tierEntry : configuredLookups.entrySet()) {
+        String tier = tierEntry.getKey();
+        Map<String, LookupStatus> lookupStatusMap = new HashMap<>();
+        result.put(tier, lookupStatusMap);
+        Collection<HostAndPort> hosts = lookupCoordinatorManager.discoverNodesInTier(tier);
+
+        for (Map.Entry<String, LookupExtractorFactoryMapContainer> lookupsEntry : tierEntry.getValue().entrySet()) {
+          lookupStatusMap.put(
+              lookupsEntry.getKey(),
+              getLookupStatus(
+                  lookupsEntry.getKey(),
+                  lookupsEntry.getValue(),
+                  hosts,
+                  lookupsStateOnNodes,
+                  detailed
+              )
+          );
+        }
+      }
+
+      return Response.ok(result).build();
+    } catch (Exception ex) {
+      LOG.error(ex, "Error getting lookups status");
```

Comment:
I feel like this kind of catch on every rest method is inefficient. Out of scope for this or, but it seems like logging the http request with error at the jetty level when all the important data is in the uri or the exception would be more sustainable